contracts/DxMgnPool.sol

```diff
@@ -135,13 +142,25 @@ contract DxMgnPool is Ownable {
         if(totalDeposit > 0){
             dx.withdraw(address(depositToken), totalDeposit);
         }
+        currentState = State.DepositWithdrawnFromDx;
     }
 
     function withdrawUnlockedMagnoliaFromDx() public {
-        require(currentState() == State.PoolingEnded, "Pooling period is not yet over.");
-        
+        require(currentState == State.DepositWithdrawnFromDx, "Unlocking not yet triggered");
+
+        // Implicit we also have:
+        // require(unlockedTokens[msg.sender].withdrawalTime < now, "The tokens cannot be withdrawn yet");
+
         mgnToken.withdrawUnlockedTokens();
         totalMgn = mgnToken.balanceOf(address(this));
+
+        currentState = State.MgnUnlocked;
+    }
+
+    function checkForStateUpdate() public {
```

Comment:
should this be an internal method?

Reply:
There is no reason to make it internal right? Also, it might be good for other to check whether they can update the state.